addons/crm/models/crm_lead.py

```diff
@@ -1549,12 +1549,14 @@ def _message_post_after_hook(self, message, msg_vals):
             # we consider that posting a message with a specified recipient (not a follower, a specific one)
             # on a document without customer means that it was created through the chatter using
             # suggested recipients. This heuristic allows to avoid ugly hacks in JS.
-            new_partner = message.partner_ids.filtered(lambda partner: partner.email == self.email_from)
+            new_partner = message.partner_ids.filtered(
+                lambda partner: partner.email == self.email_from or (self.email_normalized and partner.email_normalized == self.email_normalized)
+            )
             if new_partner:
                 self.search([
                     ('partner_id', '=', False),
-                    ('email_from', '=', new_partner.email),
-                    ('stage_id.fold', '=', False)]).write({'partner_id': new_partner.id})
+                    ('email_normalized', '=', new_partner[0].email_normalized),
```

Comment:
The test above could be truthy  based on raw email comparison, are we sure that we always have an email_normalized here?

(Same question for others)